src/renderer/components/ContextMenu.tsx

```diff
@@ -147,27 +143,68 @@ export function ContextMenu(props: {
   openCallback: (el: HTMLDivElement | null) => void
   closeCallback: () => void
 }) {
-  const menuRef = useRef<HTMLDivElement>(null)
   const didOpen = useRef<boolean>(false)
+  // References to each level menu element
+  const menuLevelEls = useRef<HTMLDivElement[]>([])
+  // Array of indices for ContextMenuItem picked on each level,
+  // always one item less than menuLevelEls
+  const [openSublevels, setSublevels] = useState<number[]>([])
+  // Which one of the last sublevel items the keyboard is focused on
+  const keyboardFocus = useRef<number>(-1)
+
+  let items = props.items.filter(val => val !== false) as ContextMenuItem[]
+
+  const levelItems: ContextMenuLevel[] = [{ items }]
+
+  for (let i = 0; i < openSublevels.length; ++i) {
+    const idx = openSublevels[i]
+
+    items = items[idx].subitems as ContextMenuItem[]
+    levelItems.push({
+      items,
+    })
+  }
+
   useLayoutEffect(() => {
-    if (!menuRef.current) {
-      throw new Error()
+    if (menuLevelEls.current.length == 0) {
+      throw new Error('Somethings wrong with menu elements')
```

Comment:
There's a mini typo here ("something IS" and I'd add a "the" before "menu elements").

Or, how about being a bit more explicit as well? For example: "No context menu elements available for display"?